internal/observability/event/event_observation.go

```diff
@@ -4,17 +4,21 @@ import (
 	"fmt"
 
 	"github.com/hashicorp/eventlogger"
+	"github.com/hashicorp/eventlogger/filters/gated"
 	"github.com/hashicorp/go-secure-stdlib/strutil"
 )
 
 // observationVersion defines the version of observation events
 const observationVersion = "v0.1"
 
 type observation struct {
-	*eventlogger.SimpleGatedPayload
-	Version     string       `json:"version"`
-	Op          Op           `json:"op,omitempty"`
-	RequestInfo *RequestInfo `json:"request_info,omitempty"`
+	Version     string                 `json:"version"`
```

Comment:
Can you comment this struct? E.g. what Header vs. Detail are, when you'd want to set Flush, etc.

Reply:
this struct is not exported and devs will never use this struct directly.